c++/src/capnp/rpc.c++

```diff
@@ -3502,6 +3500,12 @@ private:
             co_return;
           }
 
+          // Make sure to mark ourselves as Disconnected so if the shutdown task fails it doesn't
+          // cause us to call shutdown() again.
+          Connected state = kj::mv(connection.get<Connected>());
+          connection.init<Disconnected>(KJ_EXCEPTION(DISCONNECTED, "Peer disconnected."));
```

Comment:
Perhaps the error message should be "RPC was disconnected due to idleness" or something?

Reply:
Or "Peer disconnected idle session."?